### PR TITLE
create cache decorator

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    def wrapper(*args_1: tuple) -> Any:
+        def save_result_in_cache(*args_2: tuple) -> bool:
+            if args_2 not in cache_dict.keys():
+                cache_dict[args_2] = func(*args_2)
+                return True
+            return False
```
Comment (ArsenPidhoretskyi) [inline app/main.py]: This function make code more harder

File: app/main.py [hunk 2]
```diff
@@ -1,6 +1,19 @@
...
+        was_calculating = save_result_in_cache(*args_1)
+        print(
+            "Calculating new result" if
+            was_calculating else "Getting from cache"
+        )
+        return cache_dict[*args_1]
```
Comment (ArsenPidhoretskyi) [inline app/main.py]:
```suggestion
        return cache_dict[args_1]
```

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (ghost) [reply app/main.py]: thanks,  it's dizzy task

**Review by ArsenPidhoretskyi — APPROVED** [review]
Well done!

**Pull request closed without merging**